### PR TITLE
Feature log sleep

### PR DESCRIPTION
**Pull request by vEpiphyte**
Log all ``time.sleep()`` calls

### CODE REVIEW

File: prawcore/rate_limit.py [hunk 1]
```diff
@@ -1,5 +1,8 @@
 """Provide the RateLimiter class."""
 import time
+import logging
```
Comment (bboe) [inline prawcore/rate_limit.py]: Can you keep the imports sorted?

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,8 @@
 _build/
 build/
 dist/
+
+# PyCharm
```
Comment (bboe) [inline .gitignore]: Can you add these without the comments in sorted order?

File: prawcore/rate_limit.py [hunk 3]
```diff
@@ -38,10 +41,12 @@ def delay(self):
         """Sleep for an amount of time to remain under the rate limit."""
         if self.next_request_timestamp is None:
             return
-        sleep_seconds = self.next_request_timestamp - time.time()
-        if sleep_seconds <= 0:
+        stime = self.next_request_timestamp - time.time()
```
Comment (bboe) [inline prawcore/rate_limit.py]: Can you keep the variable names as `sleep_seconds`?

File: prawcore/rate_limit.py [hunk 4]
```diff
@@ -38,10 +41,12 @@ def delay(self):
...
+        if stime <= 0:
             return
-        time.sleep(sleep_seconds)
+        msg = 'Sleeping: {:0.2f} seconds prior to call'.format(stime)
```
Comment (bboe) [inline prawcore/rate_limit.py]: Can you either do away with this variable, or call it `message`? I'm a little pedantic about not shortening variable names.

File: prawcore/sessions.py [hunk 5]
```diff
@@ -55,9 +55,10 @@ def _log_request(data, method, params, url):
     def _retry_sleep(retries):
         if retries < 3:
             base = 0 if retries == 2 else 2
-            sleep_time = base + 2 * random.random()
-            log.debug('Sleeping: {:0.2f} seconds'.format(sleep_time))
-            time.sleep(sleep_time)
+            stime = base + 2 * random.random()
+            msg = 'Sleeping: {:0.2f} seconds prior to retry'.format(stime)
```
Comment (bboe) [inline prawcore/sessions.py]: Can you please only change the message and not the variable names?

File: prawcore/rate_limit.py (hunk 1 above)
Comment (vEpiphyte) [reply prawcore/rate_limit.py]: sort by alpha - will do

File: .gitignore (hunk 2 above)
Comment (vEpiphyte) [reply .gitignore]: can do

File: prawcore/rate_limit.py (hunk 3 above)
Comment (vEpiphyte) [reply prawcore/rate_limit.py]: yup - can retain the variable names. Do you want them both unified to ``sleep_seconds``?

Comment (bboe) [reply prawcore/rate_limit.py]: Yeah that seems to make more sense than the other `sleep_time`. Good catch.

File: prawcore/rate_limit.py (hunk 4 above)
Comment (vEpiphyte) [reply prawcore/rate_limit.py]: ``message`` is fine.

**Comment by bboe** [comment]
Thanks for the update, and the PR. Great work! I'll look into making a release this evening.

**Pull request merged**